exercise/exercise_models.py

```diff
@@ -443,29 +443,31 @@ def one_has_access(self, students, when=None):
         in consideration.
         """
         timing,d = self.get_timing(students, when or timezone.now())
+
+        formatted_time = date_format(d, "DATETIME_FORMAT")
```

Comment:
It looks like this messes up time zones. The message is printed in the normalized UTC+0 time zone even if the server is set for the Finnish UTC+2 time zone.

Reply:
You can see it when you submit to an exercise during the late time. I wrote the hours 17:00 but the message shows 15:00.